python/lsst/ap/association/utils.py

```diff
@@ -182,23 +199,26 @@ def convertTableToSdmSchema(apdbSchema, sourceTable, tableName):
         A table with the correct schema for the APDB and data copied from
         the input ``sourceTable``.
     """
+    if sourceTable.empty:
+        make_empty_catalog(apdbSchema, tableName)
     table = apdbSchema[tableName]
 
     data = {}
     nSrc = len(sourceTable)
 
     for columnDef in table.columns:
+        dtype = column_dtype(columnDef.datatype, nullable=columnDef.nullable)
         if columnDef.name in sourceTable.columns:
-            dtype = column_dtype(columnDef.datatype)
-            data[columnDef.name] = pd.Series(sourceTable[columnDef.name], dtype=dtype)
+            data[columnDef.name] = pd.Series(sourceTable[columnDef.name], dtype=dtype,
+                                             index=sourceTable.index)
         else:
-            dataInit = np.zeros(nSrc, dtype=column_dtype(columnDef.datatype))
             if columnDef.nullable:
                 try:
-                    dataInit.fill(None)
-                except (TypeError, ValueError):
-                    pass
-            data[columnDef.name] = pd.Series(dataInit, index=sourceTable.index)
+                    data[columnDef.name] = pd.Series([pd.NA]*nSrc, dtype=dtype, index=sourceTable.index)
+                except TypeError:
+                    data[columnDef.name] = pd.Series([np.nan]*nSrc, dtype=dtype, index=sourceTable.index)
```

Comment:
I'm not sure that `dax_apdb` can handle `pd.NA` correctly, let me check. I understand this is used as `NULL` value for nullable integer columns?

Reply:
Yes. That is important because we want to set e.g. `diaSourceId` to `NULL` for anything associated with a solar system object, and `ssSourceId` to `NULL` for anything that is not.